### ISSUE TITLE
Invalid characters in generated go code

### ISSUE THREAD

**Issue by LeeSmet** [Jumpscale/go-raml#306]
Notice the `?` characters in generated code. 

Example:
```go
package itsyouonline 
 
import ( 
    "gopkg.in/validator.v2" 
) 

type Authorizationphonenumbers? struct { 
     
        Type   []AuthorizationMap `json:"type" validate:"nonzero"`  
} 
 
 
func (s Authorizationphonenumbers?) Validate() error { 
     
    return validator.Validate(s) 
} 
```

**Comment by iwanbk** [comment]
It is bad, where is the .raml?

**Comment by LeeSmet** [comment]
Raml used to produce this code (and more)
https://github.com/itsyouonline/identityserver/blob/0.9.13/specifications/api/itsyouonline.raml

**Comment by iwanbk** [comment]
I see the issue, new feature developed at #96 doesn't handle `?` suffix yet

**Comment by iwanbk** [comment]
Fixed by #311 

However, i think your .raml is invalid.
You need to fix it this way
```diff
--- itsyouonline.ori.raml	2017-06-17 09:27:03.000000000 +0700
+++ itsyouonline.raml	2017-06-17 08:23:35.000000000 +0700
@@ -58,21 +58,17 @@
           description: The organization that is granted read access to some of the user's information
         addresses?:
           description: Mapping of the labels of the granted ones to the labels the organization requested.
-          properties:
-             type: AuthorizationMap[]
+          type: AuthorizationMap[]
         publicKeys?: AuthorizationMap[]
         emailaddresses?:
           description: Mapping of the labels of the granted ones to the labels the organization requested.
-          properties:
-             type: AuthorizationMap[]
+          type: AuthorizationMap[]
         phonenumbers?:
           description: Mapping of the labels of the granted ones to the labels the organization requested.
-          properties:
-             type: AuthorizationMap[]
+          type: AuthorizationMap[]
         bankaccounts?:
           description: Mapping of the labels of the granted ones to the labels the organization requested.
-          properties:
-             type: AuthorizationMap[]
+          type: AuthorizationMap[]
         facebook?: boolean
         github?: boolean
         organizations:
```
